File: python_converted/01_low_complexity/src/core/interfaces/types/npc/affinity.py
from typing import Any
from enum import Enum



class RelationshipType(Enum):
    STRANGER = 'stranger'
    ACQUAINTANCE = 'acquaintance'
    FRIEND = 'friend'
    CLOSE_FRIEND = 'close_friend'
    RIVAL = 'rival'
    ENEMY = 'enemy'
class NPCAffinity:
    npcId1: str
    npcId2: str
    score: float
    lastInteraction: Date
    interactionCount: float
    relationship: \'RelationshipType\' 

File: app/frontend/src/core/interfaces/types/npc/affinity.ts
export enum RelationshipType {
  STRANGER = 'stranger',
  ACQUAINTANCE = 'acquaintance',
  FRIEND = 'friend',
  CLOSE_FRIEND = 'close_friend',
  RIVAL = 'rival',
  ENEMY = 'enemy',
}

export interface NPCAffinity {
  npcId1: string;
  npcId2: string;
  score: number;
  lastInteraction: Date;
  interactionCount: number;
  relationship: RelationshipType;
} 